pennylane_qiskit/converter.py

```diff
@@ -49,6 +49,36 @@ def _check_parameter_bound(param: Parameter, unbound_params: Dict[Parameter, Any
         raise ValueError(f"The parameter {param} was not bound correctly.".format(param))
 
 
+def _expected_parameters(quantum_circuit):
+    """Gets the expected parameters and a string of their names from the QuantumCircuit.
+    Primarily serves to change a list of Parameters and ParameterVectorElements into a list
+    of Parameters and ParameterVectors. I.e.:
+
+    [Parameter('a'), ParameterVectorElement('v[0]'), ParameterVectorElement('v[1]'), ParameterVectorElement('v[2]')]
+
+    becomes [Parameter('a'), ParameterVector(name='v', length=3)].
+
+    Returns:
+        expected_params: The reorganized list of Parameters, containing Parameter and ParameterVector
+        param_name_string: a string listing the parameter names, i.e. in the example above, 'a, v'
+
+    """
+
+    # note we can't use `set` because it reorders parameters randomly
```

Comment:
The `dict` object maintains insertion order, you might want to try using that instead. The keys would be the parameter names and the values would be the parameters themselves.